gcsfs/tests/test_core.py

```diff
@@ -733,3 +734,16 @@ def test_attrs(token_restore):
         with pytest.raises(KeyError):
             gcs.getxattr(a, "foo")
         assert gcs.getxattr(a, "something") == "not"
+
+
+@my_vcr.use_cassette(match=["all"])
+def test_request_user_proejct(token_restore):
+    with gcs_maker():
+        gcs = GCSFileSystem(TEST_PROJECT, token=GOOGLE_TOKEN, user_project=TEST_PROJECT)
```

Comment:
Currently, if `user_project=None` but `project` is set,  `user_project` will be set to `project`.

It might be good to test the case where `project` and `user_project` are different.